src/test/scala/ai/privado/testfixtures/PrivadoBaseTestFixture.scala

```diff
@@ -0,0 +1,36 @@
+package ai.privado.testfixtures
+
+import ai.privado.cache.{AppCache, AuditCache, DataFlowCache, PropertyFilterCache, RuleCache, S3DatabaseDetailsCache}
+import ai.privado.entrypoint.PrivadoInput
+import ai.privado.languageEngine.base.processor.BaseProcessor
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach, Inside}
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+import scala.collection.mutable
+
+class PrivadoBaseTestFixture[T <: TestCpg](testCpgFactory: () => T)
+    extends AnyWordSpec
+    with Matchers
+    with BeforeAndAfterAll
+    with BeforeAndAfterEach
+    with Inside {
+
+  private val cpgs = mutable.ArrayBuffer.empty[TestCpg]
+
+  def code(code: String): T = {
+    val newCpg = testCpgFactory().moreCode(code)
+    cpgs.append(newCpg)
```

Comment:
Can this lead to holding a lot of memory?

Reply:
Unit tests shouldn't code samples shouldn't end up creating too much of memory footprint.

Reply:
As these will be references, should be ok